http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/AppRoutes.scala

```diff
@@ -141,92 +132,88 @@ class AppRoutes(kubernetesService: AppService[IO], userInfoDirectives: UserInfoD
           val tags = Map("appType" -> req.appType.toString)
           metrics.incrementCounter("createApp", 1, tags)
       }
-      _ <- ctx.span.fold(apiCall)(span => spanResource[IO](span, "createApp").use(_ => apiCall))
+      _ <- withSpanResource("createApp",
+                            kubernetesService.createApp(
+                              userInfo,
+                              CloudContext.Gcp(googleProject),
+                              appName,
+                              req
+                            )
+      )
     } yield StatusCodes.Accepted
 
   private[api] def getAppHandler(userInfo: UserInfo, googleProject: GoogleProject, appName: AppName)(implicit
     ev: Ask[IO, AppContext]
-  ): IO[ToResponseMarshallable] =
+  ): IO[ToResponseMarshallable] = {
+    val apiCallName = "getApp"
     for {
-      ctx <- ev.ask[AppContext]
-      apiCall = kubernetesService.getApp(
-        userInfo,
-        CloudContext.Gcp(googleProject),
-        appName
+      _ <- metrics.incrementCounter(apiCallName)
+      resp <- withSpanResource(apiCallName,
```

Comment:
can potentially put the metrics line into the `withSpanResource` too...

Reply:
I was thinking of that. However, not all routes have a metrics call like this.  Sometimes metrics are called from the service instead.

Reply:
see also `createAppHandler()` where the metrics have some associated logic

Reply:
ah right, I think for some, we needed additional info, so we had to move it to service layer...